libr/bin/format/mach0/coresymbolication.c

```diff
@@ -261,15 +271,20 @@ RCoreSymCacheElement *r_coresym_cache_element_new(RBinFile *bf, RBuffer *buf, ut
 				break;
 			}
 			ut64 sect_name_off = r_read_ble (cursor, false, bits);
+			if (!i && !sect_name_off) {
+				relative_to_strings = true;
+			}
 			cursor += word_size;
 			if (bits == 32) {
 				cursor += word_size;
 			}
-			sect->name = str_dup_safe (b, sect_start + (size_t)sect_name_off, end);
+			if (relative_to_strings) {
```

Comment:
ternary operator, call the function once, just make the argument conditional (just as a separate variable passed to the str_dup_safe thing).

Also, is there any reason to have all thoe str_dup_ helpers? looks to me that the code can directly call r_str_ndup()

Reply:
the reason for `str_dup_` helpers is that i don't know the length of the strings ahead of time, but i know they are zero-terminated and the helper just avoids me to duplicate the code needed to:
- check the start of string is inside the boundaries of the origin buffer
- calculate the length
- actually call `r_str_ndup`

Reply:
Kk that sounds fair enough :)